### PR TITLE
Store additional track/album/artist information from recently played tracks

### PR DESCRIPTION
**Pull request by Zindre17**

### CODE REVIEW

File: dbsetup.sql [hunk 1]
```diff
@@ -11,9 +11,34 @@ create table if not exists tokens (
...
+create table if not exists tracks (
+    id string primary key,
+    name string not null,
+    albumId string not null references albums(id),
+    duration int not null
+);
+
+create table if not exists trackArtists (
+    trackId string references tracks(id),
+    artistId string references artists(id),
```
Comment (adriansberg) [inline dbsetup.sql:35]: What happens in the case a track has more than one artist?

File: dbsetup.sql [hunk 2]
```diff
@@ -11,9 +11,34 @@ create table if not exists tokens (
...
+);
+
+create table if not exists albums (
+    id string primary key,
+    name string not null,
+    imageUrl string not null
+);
+
+create table if not exists tracks (
+    id string primary key,
```
Comment (adriansberg) [inline dbsetup.sql:27]: I guess multiple tracks can have the same name, should there be an autogenerated ID as PK? Same goes for album and artist.

**Review by adriansberg — APPROVED** [review]
Looks great! Only have that one question regarding multiple artists. Not a blocker.

File: dbsetup.sql (hunk 1 above)
Comment (Zindre17) [reply dbsetup.sql:35]: That is exactly what this enables :D 
So one row per artist with same trackid but different artistids

Comment (adriansberg) [reply dbsetup.sql:35]: Right, of course.

**Pull request merged**